### PR TITLE
feat/catalog-search-field: Add search field support

### PR DESCRIPTION
**Pull request by ayaanqui**
### Features to implement

- [x] Add "All" badge for "All Available Strains"
  - [x] Update tests
- [x] Descriptor autocomplete
- [x] Check GraphQL API implementation for descriptor

### CODE REVIEW

**Review by cybersiddhu — CHANGES_REQUESTED** [review]
* The search do not search anything.

**Review by cybersiddhu — CHANGES_REQUESTED** [review]
Great work 👏 so far.
Just disregard my previous comment, there are couple of tweaks that needed to be done and i will  explained it visually.

**Comment by ayaanqui** [comment]
# Updates to the search

- [x] Remove "No options" dropdown after [ENTER]
- [x] Fix descriptor so that when you close the filter it goes back to search with active filter
- [x] Add search filter to URI

**Comment by ayaanqui** [comment]
It seems like the reason why infinite scroll doesn't seem to work has something to do with how we make the GraphQL API call when a search is made, and then when it's reset, it doesn't go back to the start state (i.e the state where infinite scroll worked, the state where there was no search being made).

**Comment by ayaanqui** [comment]
> It seems like the reason why infinite scroll doesn't seem to work has something to do with how we make the GraphQL API call when a search is made, and then when it's reset, it doesn't go back to the start state (i.e the state where infinite scroll worked, the state where there was no search being made).

In addition to the above, I've seen that the scroll bug only occurs when searching for a result that does not have a next cursor (i.e. it does not have any more pages to load), like searching for `myoB` or `HL501/X55` and clearing the search result should not allow the infinite load trigger. However, results like `act15` or `GWDI` that have more than 1 page do not have this issue. As to why this is occurring, I have yet to figure out.

**Comment by ayaanqui** [comment]
> > It seems like the reason why infinite scroll doesn't seem to work has something to do with how we make the GraphQL API call when a search is made, and then when it's reset, it doesn't go back to the start state (i.e the state where infinite scroll worked, the state where there was no search being made).
> 
> In addition to the above, I've seen that the scroll bug only occurs when searching for a result that does not have a next cursor (i.e. it does not have any more pages to load), like searching for `myoB` or `HL501/X55` and clearing the search result should not allow the infinite load trigger. However, results like `act15` or `GWDI` that have more than 1 page do not have this issue. As to why this is occurring, I have yet to figure out.

@cybersiddhu After digging in some more looks like this exact issue is also present in [https://dictycr.org/stockcenter/strains?filter=bacterial](https://dictycr.org/stockcenter/strains?filter=bacterial) as well. If you start in "Regular Strains" and then look up `myoB` (which like mentioned above does not have a "next" cursor for the infinite loader) and then switch to any other strain category the infinite loader does not get triggered. Looks to me like this issue was always there and we need to find the bug and fix it.

**Comment by cybersiddhu** [comment]
Very very good observation @ayaanqui. So, lets recap the issue that i have observed...
* If you just go to any page(bacterial/regular/gwdi) the infinite scroll works. 
* If you do a chip based search, the search result does not scroll. Of course, it will not scroll if you do not have enough result to scroll, but that's not the issue.
* After you close/remove the chip, the search result does not scroll.
* Even after that the scroll is totally hosed, i think it starts to behave erratically. 
So, in short we have issue with search result, the state of the search result prevents scrolling. Now since the scrolling works when you switch the dropdown(regular to gwdi), it means the problem is associated with the part that adds or removes search based chip. So, either that piece of code is not doing the right thing or it is doing the right thing however the rest of the code is not behaving the way it is expected to do. In the later case the existing code needs to be fixed.

**Comment by ayaanqui** [comment]
@cybersiddhu I was able to find the exact problem which comes from `src/common/hooks/useLoadMoreItems.ts`. The cause of the problem is that `useLoadMoreItems` returns the callback `loadMoreItems` and the state `hasMore`. Now the problem comes in when `hasMore` is set to `false` when the `nextCursor` from the API is `0`, and then when the search has cleared the state from `useLoadMoreItems` is still set to `false` preventing the infinite scroll load more feature from triggering.

**Comment by cybersiddhu** [comment]
> @cybersiddhu I was able to find the exact problem which comes from `src/common/hooks/useLoadMoreItems.ts`. The cause of the problem is that `useLoadMoreItems` returns the callback `loadMoreItems` and the state `hasMore`. Now the problem comes in when `hasMore` is set to `false` when the `nextCursor` from the API is `0`, and then when the search has cleared the state from `useLoadMoreItems` is still set to `false` preventing the infinite scroll load more feature from triggering.

The observation is correct, how the solution is inverted because you are setting `hasMore` inside the callback `loadMoreItems`. The solution should have been that `loadMoreItems` should be triggered whenever `hasMore` is true(means nextCursor is empty). It means the checking of nextCursor has to be done outside of `loadMoreItems` and then `loadMoreItems` should be fed that value or needs to have access to it. It should not use `useHasMore` to set `hasMore` value. It should act based on that value. Now though it seems very very simple, it does needs bunch of subtle changes, particularly which component will be responsible for setting what value and how it orchestrates fetching of data using infinite scroll properly. In our current code, the responsibility is not uniformly distributed and writing more on top it just makes it even difficult to handle and maintain.

**Comment by cybersiddhu** [comment]
@Mergifyio update

**Comment by cybersiddhu** [comment]
@Mergifyio update

**Comment by cybersiddhu** [comment]
@Mergifyio update
